### ISSUE TITLE
"geowave-*-accumulo" post-install script fails inappropriately on "type hadoop" check

### ISSUE THREAD

**Issue by ewilson-radblue** [ngageoint/geowave#414]
Problem is here:
https://github.com/ngageoint/geowave/blob/master/deploy/packaging/rpm/centos/6/SOURCES/deploy-geowave-to-hdfs.sh#L13

Commands run:
sudo rpm -Uvh http://s3.amazonaws.com/geowave-rpms/release/noarch/geowave-repo-1.0-3.noarch.rpm
sudo yum --enablerepo=geowave install geowave-0.8.7-cdh5-accumulo

It fails on line 13 of "deploy-geowave-to-hdfs.sh" when checking for the Hadoop command, but I have verified that the command is available.  In fact, I'm able to run the script manually immediately afterwards using "sudo /usr/local/geowave-0.8.7-cdh5/accumulo/deploy-geowave-to-hdfs.sh".

**Comment by spohnan** [comment]
Thanks for the report on the problem. I wasn't able to reproduce given the following versions and commands. Is your environment CentOS 6.x and CDH5.x ?

## Test Setup

#### Hadoop Vendor 
    [root@c3-master ~]# cat /etc/yum.repos.d/cloudera-manager.repo
    [cloudera-manager]
    name = Cloudera Manager, Version 5.4.1

#### Operating System
    [root@c3-master ~]# cat /etc/redhat-release
    CentOS release 6.6 (Final)


## Commands Run

    sudo rpm -Uvh http://s3.amazonaws.com/geowave-rpms/release/noarch/geowave-repo-1.0-3.noarch.rpm
    sudo yum --enablerepo=geowave install geowave-0.8.7-cdh5-accumulo

## Output 

Install ran without error and iterator was present in HDFS afterwards

    ... Truncated a bit
    Installed:
      geowave-0.8.7-cdh5-accumulo.noarch 0:0.8.7-201505292113

    Dependency Installed:
      geowave-0.8.7-cdh5-tools.noarch 0:0.8.7-201505292113
      geowave-core.noarch 0:0.8.7-201505292121

    Complete!

    [root@c3-master ~]# sudo -u hdfs hadoop fs -ls /accumulo/classpath/geowave/0.8.7-cdh5
    Found 2 items
    -rw-r--r--   1 accumulo accumulo        535 2015-05-30 00:29 /accumulo/classpath/geowave/0.8.7-cdh5/geowave-accumulo-build.properties
    -rw-r--r--   1 accumulo accumulo  126936336 2015-05-30 00:28 /accumulo/classpath/geowave/0.8.7-cdh5/geowave-accumulo.jar

## Command location and user paths

hadoop is linked into /usr/bin using alternatives so it should be on the path of any user

    [root@c3-master ~]# type hadoop
    hadoop is /usr/bin/hadoop

    [root@c3-master ~]# echo $PATH
    /usr/local/sbin:/sbin:/bin:/usr/sbin:/usr/bin:/root/bin

    [spohnae@c3-master ~]$ echo $PATH
    /usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/sbin:/home/spohnae/bin

**Comment by chrisbennight** [comment]
I think Rich mentioned this was happening on RHEL (not sure what version)

**Comment by ewilson-radblue** [comment]
I encountered the issue on RHEL 6.5 and CentOS 6.5 in AWS VMs.  For environment, I don't have a Cloudera Manager installed and the Hadoop command is not linked in /usr/bin, though it is in the PATH and "type hadoop" returns the correct path.  I'll try linking it in /usr/bin and see if that changes anything.

**Comment by ewilson-radblue** [comment]
Still getting the error with the link created in /usr/bin.

Output is:

    Running Transaction
      Installing : geowave-core-0.8.7-201505292121.noarch                                                          1/3
      Installing : geowave-0.8.7-cdh5-tools-0.8.7-201505292113.noarch                                              2/3
      Installing : geowave-0.8.7-cdh5-accumulo-0.8.7-201505292113.noarch                                           3/3
    Non-fatal POSTIN scriptlet failure in rpm package geowave-0.8.7-cdh5-accumulo-0.8.7-201505292113.noarch
    warning: %post(geowave-0.8.7-cdh5-accumulo-0.8.7-201505292113.noarch) scriptlet failed, exit status 1
      Verifying  : geowave-core-0.8.7-201505292121.noarch                                                          1/3
      Verifying  : geowave-0.8.7-cdh5-accumulo-0.8.7-201505292113.noarch                                           2/3
      Verifying  : geowave-0.8.7-cdh5-tools-0.8.7-201505292113.noarch                                              3/3
    
    Installed:
      geowave-0.8.7-cdh5-accumulo.noarch 0:0.8.7-201505292113
    
    Dependency Installed:
      geowave-0.8.7-cdh5-tools.noarch 0:0.8.7-201505292113           geowave-core.noarch 0:0.8.7-201505292121

**Comment by spohnan** [comment]
We're verified the OS is the same but we have noted differences with the Hadoop setup. I've only tested installing the RPMs after using one of the CDH managed install methods - http://www.cloudera.com/content/cloudera/en/documentation/core/latest/topics/cm_ig_intro_to_cm_install.html 

I've used both the downloadable install script and also have used Puppet to do the intial configuration of the CDH agents and then used the setup wizard to install Zookeeper, HDFS and Accumulo.

How would we recreate the install procedure you've used to get to this error state?
